### PR TITLE
14 be character creation

### PR DESCRIPTION
**Pull request by adrsch04**
### Anything Else?

[//]: # (for a sample see: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/)

### CODE REVIEW

File: fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/controller/PlayerController.java [hunk 1]
```diff
@@ -1,23 +1,56 @@
...
 @RequiredArgsConstructor
 public class PlayerController {
 
-	private final PlayerSession playerSession;
+    private final PlayerSession playerSession;
+    private final CharacterService characterService;
+
+    /**
+     * @return the current player session
+     */
```
Comment (Nilstrieb) [inline fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/controller/PlayerController.java:24]: Is this comment useful?

File: fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/character/Character.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package ch.bbw.fabbwled.lands.character;
+
+import lombok.With;
+import java.util.List;
+public class Character {
+    @With
+    public record CharacterDto(String name,
```
Comment (keykey7) [inline fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/character/Character.java:7]: there is some overlap/conflicts still with what belongs to the `PlayerDto` and what belongs to `CharacterDto`. For example `name` exists in both objects, but means the same. `description` is only useful lore needed during character creation and not used later while playing the game.
How about the following:
```
public record CharacterCreateDto(PlayerDto player, String description);

public record PlayerDto(String name,
                            SectionId currentSection,
                            Set<String> titlesAndHonours,
                              RankEnum rank,
							  ProfessionEnum profession,
                               int stamina,
                               StatsDto baseStats,
                               List<String> posessions)
```
...it is then up to the FE to display what they want during character creation.

**Review by keykey7 — COMMENTED** [review]
very nice, going into the right direction!

File: fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/character/Character.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+                               RankEnum rank,
+							ProfessionEnum profession,
```
[the remark this answers is not in the document]
Comment (keykey7) [reply fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/character/Character.java:9]: export codestyles to an `.editorconfig` file (including all 3 million intellij custom settings) to change the discussion from "I think this is ugly" to a "we agreed on ..., aka code is law".

File: fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/controller/PlayerController.java [hunk 4]
```diff
@@ -1,23 +1,56 @@
...
+import java.util.Collections;
+import java.util.List;
 
 @Validated
 @RestController
 @RequiredArgsConstructor
 public class PlayerController {
 
-	private final PlayerSession playerSession;
+    private final PlayerSession playerSession;
```
Comment (keykey7) [inline fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/controller/PlayerController.java:20]: achtung hier wurden tabs zu spaces... mir ist das egal, aber wir sollten viele merge issues vermeiden. evtl macht es sinn ein `.editorconfig` file zu exportieren, damit alle mit denselben settings arbeiten. @Nilstrieb 
https://www.jetbrains.com/help/idea/editorconfig.html#53e38cbf

Comment (Nilstrieb) [reply fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/controller/PlayerController.java:20]: @adrsch04 wird sich das anschauen

**Review by Nilstrieb — APPROVED** [review]

**Review by Lion-alt — APPROVED** [review]
Looks good with added .editorignore

**Review by MagischerWolf05 — APPROVED** [review]
👍

**Review by Desi7 — APPROVED** [review]
Coding looks good 👍 next time pls write infos in the PR description

File: fabbwled-backend/.editorconfig [hunk 5]
```diff
@@ -0,0 +1,1214 @@
...
+ij_yaml_align_values_properties = do_not_align
+ij_yaml_autoinsert_sequence_marker = true
+ij_yaml_block_mapping_on_new_line = false
+ij_yaml_indent_sequence_value = true
+ij_yaml_keep_indents_on_empty_lines = false
+ij_yaml_keep_line_breaks = true
+ij_yaml_sequence_on_new_line = false
+ij_yaml_space_before_colon = false
+ij_yaml_spaces_within_braces = true
+ij_yaml_spaces_within_brackets = true
```
Comment (Nilstrieb) [inline fabbwled-backend/.editorconfig:1214]: can you put this into a separate PR?

**Review by Rwael — APPROVED** [review]
Wow, impressive! Good Tests as well 👍
